Mini blog post on providers (#2400)
pulumi/pulumi-hugo: themes/default/content/blog/dotnet-custom-providers/index.md

---
title: "Preview of .NET Core resource providers"
authors: ["fraser-waters"]
tags: ["pulumi-news", ".net", "c#", "packages", "features", "native-providers"]
meta_desc: "Today we are releasing preview support for writing Pulumi providers in any .NET Core language, including C#, F#, and VB.NET."
date: "2023-01-18"

meta_image: "pulumi-dotnet.png"
---

> Today we are pleased to announce the Preview of .NET Core support for custom resource providers. This means you can build custom providers, using your favorite .NET language, including C#, F#, and VB.NET.

Reviewer: I think that first comma might be unnecessary. 

```suggestion
Today we are pleased to announce the Preview of .NET Core support for custom resource providers. This means you can build custom providers using your favorite .NET language, including C#, F#, and VB.NET.
```

<!--more-->

> This is a first look preview release of the provider interface for .NET, it has minimal documentation but is feature complete. We hope to gather early feedback to refine the API and quickly bring this to a production quality release.

Reviewer: Maybe we need a conjunction after the ','.

> [..] of the provider interface for .NET, therefore it has [..]

But I guess that implies that "first look" = "minimal documentation."
I think a separate sentence might work better.

```suggestion
This is a first look preview release of the provider interface for .NET. It has minimal documentation but is feature complete. We hope to gather early feedback to refine the API and quickly bring this to a production quality release.
```

> You can see an example of using the new provider interface in the [dotnet integration tests](https://github.com/pulumi/pulumi-dotnet/tree/main/integration_tests/testprovider) which has replaced the old test provider which was written in Go.

Reviewer: Lots of "which"es in this sentence.
Consider:

```suggestion
You can see an example of using the new provider interface in the [dotnet integration tests directory](https://github.com/pulumi/pulumi-dotnet/tree/main/integration_tests/testprovider). This has replaced the old test provider that was written in Go.
```

---

> title: "Preview of .NET Core resource providers"

Reviewer 2: The use of `Core` died after .NET Core v3.1, everything now is just .NET
